docs/deployment-guide.md

```diff
@@ -2,16 +2,149 @@
+
+```bash
+git clone https://github.com/Alfresco/alfresco-ansible-deployment.git
+cd alfresco-ansible-deployment
+```
+
+> You may want to generate an SSH key pair locally and use it later for
+> deployment. Wether you generate one or you use one you copied over to the
+> control node, it is your responsibility to deploy it to the target machines so
+> Ansible can use it. Using SSH keys is recommended but not mandatory. If using
+> password instead make sure to add the `-k`switch to the ansible command so it
+> prompts you for a password.
+
+### Setup runtime environment
+
+Before starting using the playbook, make sure you are running at least python 3.8:
+
+```bash
+python3 --version
+pip install --user pipenv
+```
+
+We made mandatory the usage of [pipenv](https://pipenv.pypa.io/en/latest/) to
+make sure that you will run the playbook with the same set of python
+dependencies we are running our integration tests.
+
+Install pipenv via pip:
+
+```bash
+pip install --user pipenv
+```
+
+If `pip` is not available, you can bootstrap pipenv using the [crude
+installation](https://pipenv.pypa.io/en/latest/install/#crude-installation-of-pipenv):
+
+```bash
+curl https://raw.githubusercontent.com/pypa/pipenv/master/get-pipenv.py | python3
+```
+
+Now you are ready to install Ansible and required runtime dependencies in a dedicated
+virtual environment managed by pipenv.
+
+Run from the playbook folder:
+
+```bash
+pipenv install --deploy
+pipenv run ansible-galaxy install -r requirements.yml
+```
+
+> When using `pipenv`, it is sufficient to prefix any ansible command with
+> `pipenv run`. We always provide command snippets in this documentation with
+> that prefix for your copy-pasting convenience. Keep in mind that you can also
+> run `pipenv shell` that will spawn a new shell that automatically assume that
+> every command is related to the current pipenv virtualenv. You can exit from
+> that shell just by using `exit` or `Ctrl+D`.
+
+If you intend to deploy an Enterprise system, create the mandatory environment
+variables that hold your Nexus credentials as shown below (replacing the values
+appropriately):
+
+```bash
+export NEXUS_USERNAME="<your-username>"
+export NEXUS_PASSWORD="<your-password>"
+```
+
+Now you have the control node setup you can [configure](#configure-your-deployment) your deployment and decide what kind of deployment you would like.
+
+To deploy everything on the control node follow the steps in the [Localhost Deployment](#localhost-deployment) section or to deploy to one or more other machines follow the steps in the [SSH Deployment](#ssh-deployment) section.
+
+If you are going to do a production deployment, please take a look at the
+mandatory [Secrets management](#secrets-management) section.
+
+Alternatively, you can add the parameter `-e autogen_unsecure_secrets=true` to
+the `ansible-playbook` command to just autogenerate secrets before running the
+playbook for the first time (remove it for the next runs).
```

Comment:
I've rewritten this but I am wondering if I could just improve the secrets-init to not bother when autogen_unsecure_secrets is enabled

Reply:
If, so we should keep it for later and capture it in a new JIRA